### PR TITLE
📚 Add RTD documentation

### PR DESCRIPTION
**Pull request by chrisjsewell**
The simplest implementation of RTD documentation!

Simply run `tox -e docs-clean` to test, and `tox -e docs-update` to try rebuilding
(obviously a maintainer will need to add it on RTD)

In doing this, I realised that the rebuild fix (#19) wasn't actually working properly.
This is because the sphinx `env` does not get re-saved after `build-finished` events.

So instead you'll see that I now read/write a JSON file in the build folder. This is also easier to inspect manually 😄

### CODE REVIEW

**Comment by Daltz333** [comment]
I'd like the rebuild fix in a separate PR, as it's an organization policy for all merges to be squashed.

**Comment by chrisjsewell** [comment]
strange policy, fair enough

**Comment by Daltz333** [comment]
It's to enforce separate of concerns, as the primary contributors to the projects are usually external.

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+master_doc = "index"
+project = "sphinxext-rediraffe"
+
+exclude_patterns = ["_build", "Thumbs.db", ".DS_Store"]
```
Comment (Daltz333) [inline docs/conf.py:4]: I don't think excluding anything but _build is necessary, right?

Comment (chrisjsewell) [reply docs/conf.py:4]: yeh this is what we use in our excutablebooks packages, but just `_build` should generally be fine.

**Comment by chrisjsewell** [comment]
> the reason we wrote all these extensions :) and possibly more

You know that myst supports all these extensions (it supports "every" sphinx extension)? I will be adding myst to the sphinx-tabs documentation at some point.
Happy to help you convert to myst 😆 (there's a package in the works 😉 https://github.com/mmcky/sphinxcontrib-rst2myst)

But enough of the sales pitch lol, I implemented your review changes

> Additionally, using sphinx-build has had an iffy chance working on windows (due to scripts sometimes not being executable)

You've had this issue recently? I don't use Windows (because its rubbish 😛) so I don't know personally, but I haven't heard of anyone having an issue with python console_script entry points.
I always felt this was a bit of a legacy thing in sphinx, left over from python2.

> I'd like to adopt our documentation style guide into this repository and eventually add our other CI checks for the documentation. See here for our style guide.

I would suggest creating a `.github` repository for this organisation, containing this information, which then gets linked in issues/PRs across all repositories, see https://github.com/executablebooks/.github


Also, fancy also releasing a v0.2.2 soon. I've already had people complaining about rebuilds failing for jupyter-book docs 😬

**Comment by Daltz333** [comment]
It's been pushed. And I'm aware of RTD CI, as we do use it. However, it's very difficult to replicate the RTD build environment exactly, so our CI exists to replicate a "normal" build. Additionally it tests other things such as spelling, image size, verifying all links are valid, valid redirects (this extension).

**Comment by chrisjsewell** [comment]
>  it tests other things such as spelling, image size, verifying all links are valid

Fair, I'll have to check it out some time 👍

**Comment by chrisjsewell** [comment]
> It's been pushed.

thanks 😄

**Comment by AustinShalit** [comment]
@Daltz333 status update on this?

**Comment by TheTripleV** [comment]
The theme should be switched to furo to match sphinxext-opengraph.
Then, after the rtd project is created, this should be be good to go.

**Pull request merged**